### ISSUE TITLE
GRPC 1.4.4 for PHP error on Mac OSX (and timeout on Linux)

### ISSUE THREAD

**Issue by bshaffer** [grpc/grpc#12123]
## MAC OSX
When calling the gRPC libraries on Mac OX, I receive the following error:

```
E0808 14:59:19.592674000 140736785834944 sync_posix.c:60]              assertion failed: pthread_mutex_lock(mu) == 0
/usr/local/bin/phpunit: line 2: 81049 Abort trap: 6           /usr/bin/env php -d allow_url_fopen=On -d detect_unicode=Off /usr/local/Cellar/phpunit/5.3.2/libexec/phpunit-5.3.2.phar "$@"
```

I've also received this error:
```
line 2: 84323 Segmentation fault: 11  /usr/bin/env php -d allow_url_fopen=On -d detect_unicode=Off /usr/local/Cellar/phpunit/5.3.2/libexec/phpunit-5.3.2.phar "$@"
```

To further confuse things, occasionally the tests pass, but a warning is thrown after execution:

```
/usr/local/bin/phpunit: line 2: 84797 Abort trap: 6           /usr/bin/env php -d allow_url_fopen=On -d detect_unicode=Off /usr/local/Cellar/phpunit/5.3.2/libexec/phpunit-5.3.2.phar "$@"
```

This error only takes place on a SPECIFIC TEST, even though there are 5 other tests that all use the GRPC extension. The specific call here is `DLP::ListInfoTypes`. I am looking into this call to find any clues as to why this one would throw an error.

**UPDATE** I also have now received this error

```
Assertion failed: (prior > 0), function gpr_ref_non_zero, file /private/tmp/pear/temp/grpc/src/core/lib/support/sync.c, line 106.
```

## Linux
These [travis builds](https://travis-ci.org/GoogleCloudPlatform/php-docs-samples/builds/262410732) install using `pecl install grpc`, but when the extension is called (the tests that run in the `dlp/` directory), the call hangs and the build times out after 10 minutes.

**Comment by bshaffer** [comment]
**Breakthrough for MAC OSX**

The error is thrown for that specific function because in that test, two GRPC calls are made. So the issue is that the GRPC connection is used twice. If I make two GRPC calls in any one PHP request, I can reproduce the error.

**Comment by stanley-cheung** [comment]
I am actively looking into this

**Comment by bshaffer** [comment]
Here are my php config options for PHP 7.1:

```
--prefix=/usr/local/Cellar/php71/7.1.8_20 --localstatedir=/usr/local/var --sysconfdir=/usr/local/etc/php/7.1 --with-config-file-path=/usr/local/etc/php/7.1 --with-config-file-scan-dir=/usr/local/etc/php/7.1/conf.d --mandir=/usr/local/Cellar/php71/7.1.8_20/share/man --enable-bcmath --enable-calendar --enable-dba --enable-exif --enable-ftp --enable-gd-native-ttf --enable-mbregex --enable-mbstring --enable-shmop --enable-soap --enable-sockets --enable-sysvmsg --enable-sysvsem --enable-sysvshm --enable-wddx --enable-zip --with-freetype-dir=/usr/local/opt/freetype --with-gd --with-gettext=/usr/local/opt/gettext --with-iconv-dir=/usr --with-icu-dir=/usr/local/opt/icu4c --with-jpeg-dir=/usr/local/opt/jpeg --with-kerberos=/usr --with-mhash --with-ndbm=/usr --with-png-dir=/usr/local/opt/libpng --with-xmlrpc --with-zlib=/usr --with-readline=/usr/local/opt/readline --without-gmp --without-snmp --with-libxml-dir=/usr/local/opt/libxml2 --with-pdo-odbc=unixODBC,/usr/local/opt/unixodbc --with-unixODBC=/usr/local/opt/unixodbc --with-bz2=/usr --with-openssl=/usr/local/opt/openssl --enable-fpm --with-fpm-user=_www --with-fpm-group=_www --with-curl --with-xsl=/usr --with-ldap --with-ldap-sasl=/usr --with-mysql-sock=/tmp/mysql.sock --with-mysqli=mysqlnd --with-pdo-mysql=mysqlnd --disable-opcache --enable-pcntl --enable-dtrace --disable-phpdbg --enable-zend-signals
```

**Comment by stanley-cheung** [comment]
Should be fixed in 1.4.6
